packages/aws-cdk-lib/aws-appsync/lib/graphqlapi.ts

```diff
@@ -575,6 +701,36 @@ export class GraphqlApi extends GraphqlApiBase {
     };
   }
 
+  private setupSourceApiAssociations() {
+    this.definition.sourceApiOptions?.sourceApis.forEach(sourceApiOption => {
+      new CfnSourceApiAssociation(this, `${sourceApiOption.sourceApi.node.id}Association`, {
+        sourceApiIdentifier: sourceApiOption.sourceApi.apiId,
+        mergedApiIdentifier: this.api.attrApiId,
+        sourceApiAssociationConfig: {
+          mergeType: sourceApiOption.mergeType ?? MergeType.AUTO_MERGE,
+        },
+      });
+    });
+  }
+
+  private setupMergedApiExecutionRole(sourceApiOptions: SourceApiOptions) {
+    if (sourceApiOptions.mergedApiExecutionRole) {
+      this.mergedApiExecutionRole = sourceApiOptions.mergedApiExecutionRole;
+    } else {
+      const sourceApiArns = sourceApiOptions.sourceApis?.map(sourceApiOption => {
+        return sourceApiOption.sourceApi.arn;
```

Comment:
We also need this role to be able to access all top level fields like: <source api arn>:*. Can you fix this to add both the source api arn alone and the source api arn with wildcard for all top level fields? https://docs.aws.amazon.com/appsync/latest/devguide/merged-api.html#execution-roles-merged-api

Reply:
Thanks for pointing this out. I will fix it next week. I'm on holiday at the moment.

Reply:
Hey, I am actually going to be in this code for an enhancement here so I can go ahead and submit these changes if you want?